### PR TITLE
Adding JAEGER_SERVICE_NAME for OpenShift S2I and fix preference service name, otherwise it does not deploy/run

### PR DESCRIPTION
**Pull request by lbroudoux**
Using OpenShift S2I build in order to deploy on OCP cluster, we must add `JAEGER_SERVICE_NAME` env to `preference` deployment.

Otherwise, embedded Tomcat is not able to start with missing service name for `tracingFilter`:

```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'tracer' defined in com.redhat.developer.demos.preference.PreferencesApplication: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.opentracing.Tracer]: Factory method 'tracer' threw exception; nested exception is java.lang.IllegalArgumentException: Service name must not be null or empty
```

### CODE REVIEW

File: docs/2deploy-microservices.adoc [hunk 1]
```diff
@@ -75,7 +75,7 @@ kubectl create -f ../../kubernetes/Service.yml -n tutorial
 
 [source, bash]
 ----
-oc new-app --name=customer --context-dir=customer/java/springboot JAVA_OPTIONS='-Xms128m -Xmx256m -Djava.net.preferIPv4Stack=true' fabric8/s2i-java~https://github.com/redhat-developer-demos/istio-tutorial -o yaml  > customer.yml
+oc new-app --name=customer --context-dir=customer/java/springboot JAVA_OPTIONS='-Xms128m -Xmx256m -Djava.net.preferIPv4Stack=true' JAEGER_SERVICE_NAME=customer fabric8/s2i-java~https://github.com/redhat-developer-demos/istio-tutorial -o yaml  > customer.yml
```
Comment (rafabene) [inline docs/2deploy-microservices.adoc]: It seems that we need the -e flag for each variable.

File: docs/2deploy-microservices.adoc [hunk 2]
```diff
@@ -160,9 +160,9 @@ kubectl create -f ../../kubernetes/Service.yml
 
 [source, bash]
 ----
-oc new-app -l app=preference,version=v1 --name=preference-v1 --context-dir=preference/java/springboot JAVA_OPTIONS='-Xms128m -Xmx256m -Djava.net.preferIPv4Stack=true' fabric8/s2i-java~https://github.com/redhat-developer-demos/istio-tutorial -o yaml  > preference.yml
+oc new-app -l app=preference,version=v1 --name=preference-v1 --context-dir=preference/java/springboot JAVA_OPTIONS='-Xms128m -Xmx256m -Djava.net.preferIPv4Stack=true' JAEGER_SERVICE_NAME=preference fabric8/s2i-java~https://github.com/redhat-developer-demos/istio-tutorial -o yaml  > preference.yml
```
Comment (rafabene) [inline docs/2deploy-microservices.adoc]: -e flag here also.

File: docs/2deploy-microservices.adoc (hunk 1 above)
Comment (lbroudoux) [reply docs/2deploy-microservices.adoc]: It seems that using `oc v3.9.27`, `-e` flag is not necessary. However we could add it for safer/cleaner command.

**Comment by lbroudoux** [comment]
Hi @rafabene ,

I have added explicit `--env` flags for setting environment variables. I have also fix and force the preference service name. Adding `JAEGER_SERVICE_NAME` for recommendation is not mandatory - at least for Vertx implementation - I think this is because there's no explicit declaration to Jaeger lib in pom.xml, every OpenTracing stuffs should be handled by envoy/istio-proxy only.

**Comment by lordofthejars** [comment]
@lbroudoux I think that these changes should be done after https://github.com/redhat-developer-demos/istio-tutorial/pull/105 is merged, so we have them on the latest bits.

**Comment by lbroudoux** [comment]
Ok, fine. I've tried yesterday deploying the tutorial on an Istio 1.0.0 on top of OCP 3.9. I've run the 4 first steps and they're OK. However, Jaeger service name enforcing seems to break stuffs in tracing and new Istio Grafana workload dashboards => customer service does not appear, recommendation service names seems broker, etc ...

Let me know if I can help...

**Comment by lordofthejars** [comment]
@lbroudoux I tried some random stuff in Jaeger and it seemed to me that it was working.  Of course, if you identify wrong parts of the configuration and you want to fix, I can give you write permissions to my repo, so you can collaborate directly to my PR.

**Comment by rafabene** [comment]
These changes were fixed by #105

**Pull request closed without merging**